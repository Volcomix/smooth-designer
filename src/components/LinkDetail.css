.LinkDetail {
  position: absolute;
  stroke: var(--text-color);
  transition: var(--ease-out-transition);
}
