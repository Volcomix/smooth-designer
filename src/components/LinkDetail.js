//@flow
import React from 'react'
import './LinkDetail.css'
import type { Block } from '../types'

export type Props = {
  fromBlock: Block,
  toBlock: Block,
}

const LinkDetail = ({ fromBlock, toBlock }: Props) => {
  const { x: x1, y: y1 } = fromBlock
  const { x: x2, y: y2 } = toBlock
  return (
    <svg
      className="LinkDetail"
      style={{
        left: Math.min(x1, x2),
        top: Math.min(y1, y2),
        width: Math.max(Math.abs(x2 - x1), 1),
        height: Math.max(Math.abs(y2 - y1), 1),
        transform: `scale(${x2 > x1 ? 1 : -1}, ${y2 > y1 ? 1 : -1})`,
      }}
    >
      <line x1={0} y1={0} x2="100%" y2="100%" />
    </svg>
  )
}

export default LinkDetail
