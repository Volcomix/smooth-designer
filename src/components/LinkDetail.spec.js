//@flow
import React from 'react'
import { shallow } from 'enzyme'
import LinkDetail from './LinkDetail'

const setup = props => {
  const wrapper = shallow(<LinkDetail {...props} />)
  return { wrapper, props }
}

it('renders without crashing', () => {
  const { wrapper } = setup({
    fromBlock: { id: '0', name: 'Block 1', x: 10, y: 30, width: 0, height: 0 },
    toBlock: { id: '1', name: 'Block 2', x: 20, y: 40, width: 0, height: 0 },
  })
  expect(wrapper).toMatchSnapshot()
})

it('renders without crashing when inverted', () => {
  const { wrapper } = setup({
    fromBlock: { id: '0', name: 'Block 1', x: 20, y: 40, width: 0, height: 0 },
    toBlock: { id: '1', name: 'Block 2', x: 10, y: 30, width: 0, height: 0 },
  })
  expect(wrapper).toMatchSnapshot()
})
