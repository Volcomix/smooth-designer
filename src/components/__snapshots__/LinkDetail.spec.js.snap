// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<svg
  className="LinkDetail"
  style={
    Object {
      "height": 10,
      "left": 10,
      "top": 30,
      "transform": "scale(1, 1)",
      "width": 10,
    }
  }
>
  <line
    x1={0}
    x2="100%"
    y1={0}
    y2="100%"
  />
</svg>
`;

exports[`renders without crashing when inverted 1`] = `
<svg
  className="LinkDetail"
  style={
    Object {
      "height": 10,
      "left": 10,
      "top": 30,
      "transform": "scale(-1, -1)",
      "width": 10,
    }
  }
>
  <line
    x1={0}
    x2="100%"
    y1={0}
    y2="100%"
  />
</svg>
`;
